scripts/migrate_data

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+"""
+Migrate .fits files whose format has changed between jwst package versions.
+"""
+import argparse
+from datetime import datetime
+import os
+import re
+import traceback
+import warnings
+
+import asdf
+from astropy.io import fits
+import numpy as np
+from packaging.specifiers import SpecifierSet
+
+import jwst
+from jwst import datamodels
+
+
+def parse_args():
+    parser = argparse.ArgumentParser("migrate_data", "migrate .fits files whose format has changed between jwst package versions")
+
+    parser.add_argument("files", nargs="+", help="one or more .fits files")
+
+    output_group = parser.add_mutually_exclusive_group(required=True)
+    output_group.add_argument("--output-dir", help="write modified files to an output directory")
+    output_group.add_argument("--in-place", help="modify files in-place", action="store_true")
+
+    return parser.parse_args()
+
+
+# If there get to be many of these we may want to move
+# them to jwst.datamodels somewhere:
+
+def rebuild_spec_table(table):
+    """
+    spectable.schema added additional columns and renamed
+    two columns.
+    """
+    schema = asdf.schema.load_schema("http://stsci.edu/schemas/jwst_datamodel/spectable.schema")
+    dtype = asdf.tags.core.ndarray.asdf_datatype_to_numpy_dtype(schema["datatype"])
+    renamed_columns = {
+        "ERROR": "FLUX_ERROR",
+        "BERROR": "BKGD_ERROR",
+    }
+
+    new_table = np.zeros(table.shape, dtype=dtype)
+    for column_name in table.dtype.names:
+        new_table[renamed_columns.get(column_name, column_name)] = table[column_name]
+
+    return new_table
+
+
+def migrate_spec_model_1_1_0(hdul):
+    for hdu in hdul:
+        if hdu.name == "EXTRACT1D":
+            hdu.data = rebuild_spec_table(hdu.data)
+
+
+def migrate_multispec_model_1_1_0(hdul):
+    for hdu in hdul:
+        if hdu.name == "EXTRACT1D":
+            hdu.data = rebuild_spec_table(hdu.data)
```

Comment:
These functions are identical.  Is there a reason they are differently-named?  Can't the `_MIGRATE_METHODS` dict below point to a single function for the two different datamodel classes?

Reply:
Hmm... good point, I thought I was going to need different logic for MultiSpecModel but they sure are identical.

Reply:
I made the exact same discovery when writing the readers for `specutils` for these a year ago.  😅